test/server.ts

```diff
@@ -237,6 +247,98 @@ test('store server started in the background should use store location wanted by
   t.notOk(await pathExists(serverJsonPath), 'server.json removed')
 })
 
+async function testParallelServerStart (options: {
+  test: tape.Test,
+  timeoutMillis?: number,
+  onProcessClosed?: (serverProcess: ChildProcess, weAttemptedKill: boolean) => void,
+  n?: number,
+  }) {
+  const pnpmBin = 'pnpm'
+  const serverProcessList: ServerProcess[] = []
+  // Variable that allows us to cancel the timeout.
+  let timeoutPromise: PDelayedPassThroughThunk<void>|null = delay(!options.timeoutMillis && options.timeoutMillis !== 0 ? 10000 : options.timeoutMillis)
+  // Promise that completes when all server processes have terminated.
+  let completedPromise: Promise<void> = Promise.resolve()
+  const n = !options.n && options.n !== 0 ? 10 : options.n
+  for (let i = 0; i < n; i++) {
+    const item: ServerProcess = [
+      crossSpawn.spawn(
+        pnpmBin,
+        ['server', 'start',
+          '--store', '../store2',
+          // We don't have to specify --port or --path explicitly because exclusive access to server.json
+          // also fixes the possibility of port collision.
+        ]),
+      createDeferred<void>(),
+      // This is true if and only if we attempted to kill the process via a signal.
+      false]
+    serverProcessList.push(item)
+    byline(item[0].stderr).on('data', (line: Buffer) => options.test.comment(`${item[0].pid}: ${line}`))
+    byline(item[0].stdout).on('data', (line: Buffer) => options.test.comment(`${item[0].pid}: ${line}`))
+    item[0].on('exit', (code: number|null, signal: string|null) => {
+      if (options.onProcessClosed) {
+        (options.onProcessClosed)(item[0], item[2])
+      }
+      for (let j = 0; j < serverProcessList.length; j++) {
+        if (serverProcessList[j][0] === item[0]) {
+          serverProcessList.splice(j, 1)
+          break
+        }
+      }
+      item[1].resolve()
+    });
+    completedPromise = completedPromise.then(() => item[1].promise)
+  }
+  await Promise.all([
+    (async () => {
+      await completedPromise
+      // Don't fire timeout if all server processes completed for some reason.
+      if (timeoutPromise !== null) {
+        timeoutPromise.cancel()
+      }
+    })(),
+    (async () => {
+      try {
+        await timeoutPromise
+      } catch (error) {
+        if (error instanceof CancelError) {
+          // All child processes completed, don't fire timeout or try to kill processes.
+          return
+        }
+        // This should never happen so it's OK if we don't kill remaining server processes in this path.
+        throw error
+      }
+      timeoutPromise = null
+      await Promise.all(serverProcessList.map(async (item) => {
+        item[2] = true
+        // Use SIGINT so that the process can delete server.json for the next test.
+        await kill(item[0].pid, 'SIGINT')
```

Comment:
This might be the reason why the test fails on AppVeyor sometimes. If you check the other tests, you'll see that we use `await execPnpm('server', 'stop')` to stop the server.

You could either
1. fix the issue on Windows, if you know how
2. ignore the `server.json` check when running the test on Windows
3. use server stop instead of killing the process